src/view_translate_bisulfite.hpp

```diff
@@ -0,0 +1,428 @@
+//                       "The inner range of the range parameter to views::translate_bisulfite must model "
+//                       "std::ranges::random_access_range.");
+//         static_assert(std::is_same_v<std::remove_const_t<seqan3::innermost_value_type_t<urng_t>>, seqan3::dna4>,
+//                       "The range parameter to views::translate_bisulfite must be over a range over elements of "
+//                       "seqan3::dna4.");
+//
+//         return view_translate_bisulfite{std::forward<urng_t>(urange)};
+//     }
+//
+//     template <std::ranges::range urng_t>
+//     constexpr friend auto operator|(urng_t && urange, translate_bisulfite_fn const & me)
+//     {
+//         return me(std::forward<urng_t>(urange));
+//     }
+// };
+
+struct translate_bisulfite_fn
+{
+    template <std::ranges::range urng_t>
+    constexpr auto operator()(urng_t && urange) const
+    {
+        static_assert(seqan3::dimension_v<urng_t> == 2,
+                      "This adaptor only handles range-of-range (two dimensions) as input.");
+        static_assert(std::ranges::viewable_range<urng_t>,
+                      "The range parameter to views::translate_bisulfite cannot be a temporary of a non-view range.");
+        static_assert(std::ranges::viewable_range<std::ranges::range_reference_t<urng_t>>,
+                      "The inner range of the range parameter to views::translate_bisulfite cannot be a "
+                      "temporary of a non-view range.");
+        static_assert(std::ranges::sized_range<urng_t>,
+                      "The range parameter to views::translate_bisulfite must model std::ranges::sized_range.");
+        static_assert(std::ranges::sized_range<std::ranges::range_reference_t<urng_t>>,
+                      "The inner range of the range parameter to views::translate_bisulfite must model "
+                      "std::ranges::sized_range.");
+        static_assert(std::ranges::random_access_range<urng_t>,
+                      "The range parameter to views::translate_bisulfite must model std::ranges::random_access_range.");
+        static_assert(std::ranges::random_access_range<std::ranges::range_reference_t<urng_t>>,
+                      "The inner range of the range parameter to views::translate_bisulfite must model "
+                      "std::ranges::random_access_range.");
+        static_assert(std::is_same_v<std::remove_const_t<seqan3::innermost_value_type_t<urng_t>>, seqan3::dna4>,
+                      "The range parameter to views::translate_bisulfite must be over a range over elements of "
+                      "seqan3::dna4.");
+
+        return std::forward<urng_t>(urange) | seqan3::views::pos_transform([] (auto && urange, size_t pos)
+        {
+            size_t index_strand = pos % 2;
+            return urange[pos] | seqan3::views::to_bisulfite_semialphabet(index_strand - 1); // We want forward strand first
```

Comment:
```suggestion
            return urange[pos] | seqan3::views::to_bisulfite_semialphabet(pos %2 == 1);
```
?

Reply:
Mine works as well, just not so nice :laughing: